lib/protest/test_case.rb

```diff
@@ -32,7 +32,9 @@ class TestCase
     # Run all tests in this context. Takes a Runner instance in order to
     # provide output.
     def self.run(runner)
-      tests.each {|test| runner.report(test) }
+      regex = Regexp.new(ARGV.first.to_s)
+      tests_to_run = tests.select { |test| "#{test.class.description} #{test.name}" =~ regex }
```

Comment:
I'd create the regexp and filter the tests only if `ARGV.first` is passed, and maybe make it a flag? Like `-m "regexp"`.